Sources/FoundationExtensions/Data/Data+Matches.swift

```diff
@@ -0,0 +1,115 @@
+// Copyright © 2023 Lautsprecher Teufel GmbH. All rights reserved.
+
+import Foundation
+
+extension Data {
+
+    public struct MatchMask: CustomStringConvertible {
+        let expected: Data
+        let mask: Data
+
+        public init(expected: Data, mask: Data? = nil) {
```

Comment:
Isn't an optional mask misleading the purpose of the model at all? Otherwise you could just use Data directly and don't need a mask at all?

Reply:
If we want to provide a default mask, I'd prefer it as a default value in the parameter so we can get rid of it's optionality 🤔

Reply:
A nil maks is turned into a series of `0xff` -  basically a full mask. The problem with default parameters is, that the default mask has to be the same length as the expected Data, so we have to figure that out at runtime.

In any case, it makes sense to use the nil mask, since `matches()` compares the _beginning_ of the given data bytes. For simple equality we can use `==`, but for consistency this is also covered by `matches()` (see the test cases).

Reply:
Makes sense